### PR TITLE
chore: fix CI cache

### PR DESCRIPTION
**Pull request by lucarin91**
#### :question: What

Configure the CI cache as suggested [here](https://github.com/actions/cache/blob/main/examples.md#go---modules).
The previous configuration restore, all the time, an empty cache.

#### :white_check_mark: Checklists

- [x] Have the changes in this PR been functionally tested?
- [x] Have the steps in [CONTRIBUTING.md](./CONTRIBUTING.md) been followed to update a Go module?

### CODE REVIEW

File: .github/workflows/win.yml [hunk 1]
```diff
@@ -17,7 +17,8 @@ jobs:
       - uses: actions/cache@v2
         with:
           path: |
-            ~/go/pkg/mod              # Module download cache
+            ~\go\pkg\mod
+            ~\AppData\Local\go-build
```
Comment (arl) [inline .github/workflows/win.yml:21]: The linux syntax above is ok, but for Windows this is not in line with what I found on https://github.com/mvdan/github-actions-golang#how-do-i-set-up-caching-between-builds:
 - the use of backslashes
 - `%LocalAppData%\go-build`

**Review by arl — COMMENTED** [review]
Look at the comments for the Windows actions

File: .github/workflows/win.yml (hunk 1 above)
Comment (lucarin91) [reply .github/workflows/win.yml:21]: For what I see in the [cache examples](https://github.com/actions/cache/blob/main/examples.md), `~\AppData\Local` and `%LocalAppData%` are the same. I think that is just a matter of taste. If you prefer I will switch to:
```
  ~\go\pkg\mod
  %LocalAppData%\go-build
```

Comment (arl) [reply .github/workflows/win.yml:21]: Yeah better using the most generic one,  %LocalAppData% then

**Review by arl — APPROVED** [review]
LGTM!

**Pull request merged**